osgar/drivers/realsense.py

```diff
@@ -85,17 +86,22 @@ def pose_callback(self, frame):
     def depth_callback(self, frameset):
         try:
             assert frameset.is_frameset()
-            frame = frameset.as_frameset().get_depth_frame()
-            n = frame.get_frame_number()
+            depth_frame = frameset.as_frameset().get_depth_frame()
+            color_frame = frameset.as_frameset().get_color_frame()
+            n = depth_frame.get_frame_number()
             if n % self.depth_subsample != 0:
                 return
-            assert frame.is_depth_frame()
-            depth_image = np.asanyarray(frame.as_depth_frame().get_data())
+            assert depth_frame.is_depth_frame() and color_frame.is_frame()
```

Comment:
I would split this into two separate asserts ... otherwise you cannot tell which one failed

Reply:
And in addition, maybe the right function to call is `is_video_frame` instead of `is_frame`. The way to confirm would be calling `depth_frame.is_frame()` and getting `True`.